arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -100,11 +100,37 @@ def errorOnRequiredPositiveFact(
         )
 
 
+def findFactsbyContextId(facts: set[ModelFact],
+                         currentContextId: str) -> Optional[ModelFact]:
+    for fact in facts:
+        if fact.context.id == currentContextId and fact.xValid >= VALID:
+            return fact
+    return None
+
+
+def getInstantContextsMatchingDurationEnd(
```

Comment:
This is also filtering based on the presence of context dimensions. The name of the function should reflect that.